lib/class-wp-rest-url-details-controller.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ *
+ *
+ * @package gutenberg
+ * @since 5.?.0
+ */
+
+/**
+ * Controller which provides REST endpoint for retrieving information
+ * from a remote site's HTML response.
+ *
+ * @since 5.?.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_URL_Details_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructs the controller.
+	 *
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'url-details';
+	}
+
+	/**
+	 * Registers the necessary REST API routes.
+	 *
+	 * @access public
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/title',
```

Comment:
What if want other fields in the future. Why not just have a details endpoint, with different fields?

Reply:
Agreed. This should be more future proof and anticipate more fields than just the titlee.
Devs would want to add additional fields, for example to also return the URL's favicon.

Reply:
I've updated to allow for this.